crates/sage-cli/src/main.rs

```diff
@@ -368,10 +431,12 @@ fn main() -> anyhow::Result<()> {
         )
         .get_matches();
 
+    // get batch-size value first, or default value, as u16 then casted back to usize
     let parallel = matches
-        .get_one::<usize>("batch-size")
+        .get_one::<u16>("batch-size")
         .copied()
-        .unwrap_or(num_cpus::get() / 2);
+        .unwrap_or_else(|| num_cpus::get() as u16 / 2) as usize;
+
```

Comment:
all of that to have a nice error message when users set `--batch-size 0` because `value_parser(usize).range(1..) is not implemented`. I am more familiar with the derive API of Clap, I am sure there are better ways to do that